doc/packages/build-support.md

```diff
@@ -0,0 +1,92 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
+
+Whereas in a build script, one might call `substitute $infile $outfile --replace-fail @foo@
+${foopkg}/bin/foo`, one might accomplish this with `pkgs.substitute` as such:
+
+```nix
+{ substitute, foopkg }:
+substitute {
+  src = ./sourcefile.txt;
+  substitutions = [
+    "--replace"
+    "@foo@"
+    "${foopkg}/bin/foo"
+  ];
+}
+```
+
+## `pkgs.substituteAll` {#pkgs-substituteall}
+
+`pkgs.substituteAll` is a wrapper around [the `substituteAll` function](#fun-substituteAll) provided
+by the stdenv. It substitutes all instances of `@varName@` in `src` with the value of environment
```

Comment:
```suggestion
by the stdenv. It substitutes all instances of `@varName@` ( `@`s included  )  in the file `src` with the value of the corresponding environment
```

Reply:
Personally I find this less clear, but I'm happy to include it -- sometimes it's hard to put myself into the shoes of a reader!